avr/cores/megacommand/MCL/SDDrivePage.cpp

```diff
@@ -0,0 +1,240 @@
+#include "SDDrivePage.h"
+#include "MCL.h"
+
+const char *c_snapshot_suffix = ".snp";
+const char *c_snapshot_root = "/SDDrive/MD";
+
+void SDDrivePage::setup() {
+  SD.mkdir(c_snapshot_root, true);
+  SD.chdir(c_snapshot_root);
+  strcpy(lwd, c_snapshot_root);
+  FileBrowserPage::setup();
+}
+
+void SDDrivePage::init() {
+
+  DEBUG_PRINT_FN();
+  md_exploit.off();
+  //  !note match only supports 3-char suffix
+  strcpy(match, c_snapshot_suffix);
+  show_dirs = true;
+  strcpy(title, "SD-Drive");
+  FileBrowserPage::init();
+}
+
+void SDDrivePage::save_snapshot() {
+  DEBUG_PRINT_FN();
+
+  MDSound sound;
+  char sound_name[] = "________";
```

Comment:
Change this to white space "       ".
The TextInputPage has code to remove trailing white space from file names.

Reply:
done